File: backend/src/main/java/model/dto/TaskDTO.java
package model.dto;

public class TaskDTO {
    private Long id;
    private String title;
    private String summary;
    private String description;
    private Integer points;
    private Boolean teamwork;
    private Integer headcount;
    private Long modulId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getPoints() {
        return points;
    }

    public void setPoints(Integer points) {
        this.points = points;
    }

    public Boolean getTeamwork() {
        return teamwork;
    }

    public void setTeamwork(Boolean teamwork) {
        this.teamwork = teamwork;
    }

    public Integer getHeadcount() {
        return headcount;
    }

    public void setHeadcount(Integer headcount) {
        this.headcount = headcount;
    }

    public Long getModulId() {
        return modulId;
    }

    public void setModulId(Long modulId) {
        this.modulId = modulId;
    }
}

File: frontend/src/app/shared/dto/task.dto.ts
export class TaskDto {
  public readonly id: number | null;
  public readonly title: string;
  public readonly summary: string;
  public readonly description: string;
  public readonly points: number;
  public readonly teamwork: boolean;
  public readonly headcount: number;
  public readonly modulId: number;

  constructor(
    id: number | null,
    title: string,
    summary: string,
    description: string,
    points: number,
    teamwork: boolean,
    headcount: number,
    modulId: number
  ) {
    this.id = id;
    this.title = title;
    this.summary = summary;
    this.description = description;
    this.points = points;
    this.teamwork = teamwork;
    this.headcount = headcount;
    this.modulId = modulId;
  }
}
